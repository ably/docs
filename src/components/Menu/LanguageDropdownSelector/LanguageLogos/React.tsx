import React from 'react';

// The best one.
export const ReactJS = () => (
  <svg width="24" height="24" viewBox="-11.5 -10.23174 23 20.46348" fill="none" xmlns="http://www.w3.org/2000/svg">
    <circle cx="0" cy="0" r="2.05" fill="#61dafb" />
    <g stroke="#61dafb" strokeWidth="1" fill="none">
      <ellipse rx="11" ry="4.2" />
      <ellipse rx="11" ry="4.2" transform="rotate(60)" />
      <ellipse rx="11" ry="4.2" transform="rotate(120)" />
    </g>
  </svg>
);
