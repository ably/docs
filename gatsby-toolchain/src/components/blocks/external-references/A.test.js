import { A } from '.';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Link } from 'gatsby';

const gatsbyRootElement = <A data={'Lorem ipsum'} attribs={{ href: 'https://www.ably.com/documentation/lorem' }}/>;
const normalRootElement = <A data={'Lorem ipsum'} attribs={{ href: 'https://www.example.com' }}/>;

describe('Different data provided to link elements results in different components', () => {
    const gatsbyRootElementRenderer = TestRenderer.create(gatsbyRootElement);
    test('Successfully renders Gatsby links', () => {
        expect(gatsbyRootElementRenderer.toJSON()).toMatchSnapshot();
    });
    test('Contains a Gatsby Link', () => {
        expect(() => gatsbyRootElementRenderer.root.findByType(Link)).not.toThrow();
    })
    const normalRootElementRenderer = TestRenderer.create(normalRootElement);
    test('Successfully renders normal links', () => {
        expect(normalRootElementRenderer.toJSON()).toMatchSnapshot();
    });
    test('Contains a normal Link', () => {
        expect(() => gatsbyRootElementRenderer.root.findByType('a')).not.toThrow();
    })
});