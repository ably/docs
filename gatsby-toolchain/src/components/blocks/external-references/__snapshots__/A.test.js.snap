// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Different data provided to link elements results in different components Successfully renders Gatsby links 1`] = `
<a
  href="https://:www.ably.com/documentation/lorem"
>
  Lorem ipsum
</a>
`;

exports[`Different data provided to link elements results in different components Successfully renders normal links 1`] = `
<a
  href="https://:www.example.com"
>
  Lorem ipsum
</a>
`;
